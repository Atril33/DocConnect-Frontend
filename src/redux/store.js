import { configureStore } from '@reduxjs/toolkit';
import authReducer from './auth/authSlice';
import doctorReducer from './doctors/doctorSlice';

export const store = configureStore({
  reducer: {
    auth: authReducer,
    doctors: doctorReducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware({
    serializableCheck: false,
  }),
});

const selectAuth = (store) => store.auth;

export {
  selectAuth,
};

export default store;
