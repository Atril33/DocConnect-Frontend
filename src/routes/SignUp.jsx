import React, { useState } from 'react';
import { useForm } from 'react-hook-form';
import { useDispatch, useSelector } from 'react-redux';
import { Link, Navigate } from 'react-router-dom';
import { toast } from 'react-toastify';
import * as Yup from 'yup';
import { yupResolver } from '@hookform/resolvers/yup';
import { selectAuth } from '../redux/store';
import { registerUser } from '../redux/auth/authActions';
import useSession from '../hooks/useSession';
import Input from '../components/Input';
import Button from '../components/Button';

const SignUp = () => {
  const [checked, setChecked] = useState(false);
  const [userSignedIn] = useSession();
  const {
    error, needsConfirmation,
  } = useSelector(selectAuth);
  const dispatch = useDispatch();
  const validationSchema = Yup.object().shape({
    name: Yup.string()
      .required('Name is required'),
    email: Yup.string()
      .required('Email is required')
      .email('Email is invalid'),
    password: Yup.string()
      .min(6, 'Password must be at least 6 characters')
      .required('Password is required'),
    password_confirmation: Yup.string()
      .oneOf([Yup.ref('password'), null], 'Passwords must match')
      .required('Confirm Password is required'),
    accept_terms: Yup.bool()
      .oneOf([true], 'Accept Ts & Ps is required'),
  });
  const formOptions = { resolver: yupResolver(validationSchema) };

  const { register, handleSubmit, formState } = useForm(formOptions);

  const { errors } = formState;

  const signup = (data) => {
    toast.promise(
      dispatch(
        registerUser(data),
      ).then(() => {
        if (error) {
          toast.error(`Oops, something went wrong: ${error}`);
        }
      }),
      {
        pending: 'loading...',
        error,
        success: 'Service is working!',
      },
    );
  };

  if (needsConfirmation) {
    return <Navigate to="/login" />;
  }

  if (userSignedIn) {
    return <Navigate to="/appointment-list" />;
  }

  return (
    <div
      className="bg-home-background-mobile bg-cover bg-center h-screen relative flex justify-center items-center md:bg-home-background"
    >
      <div className="w-full max-w-xs">
        <h1 className="text-white text-center mb-6 text-5xl font-bold">Sign Up</h1>
        <form className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4" onSubmit={handleSubmit(signup)}>
          <div className="mb-4">
            <Input register={register} name="name" placeholder="Username" type="text" />
            <div className="text-green-700 text-xs">
              {errors.name?.message}
            </div>
          </div>
          <div className="mb-4">
            <Input register={register} name="email" placeholder="Email" type="email" />
            <div className="text-green-700 text-xs">
              {errors.email?.message}
            </div>
          </div>
          <div className="mb-4">
            <Input
              register={register}
              name="password"
              placeholder="Password"
              type="password"
            />
            <div className="text-green-700 text-xs">
              {errors.password?.message}
            </div>
          </div>
          <div className="mb-6">
            <Input
              register={register}
              name="password_confirmation"
              placeholder="Confirm password"
              type="password"
            />
            <div className="text-green-700 text-xs">
              {errors.password_confirmation?.message}
            </div>
          </div>
          <div className="flex items-center mb-4">
            <input
              checked={checked}
              id="checked-checkbox"
              type="checkbox"
              {...register('accept_terms')}
              onChange={(e) => {
                setChecked(e.target.checked);
              }}
              className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-700 dark:border-gray-600"
            />
            <div>
              <label htmlFor="checked-checkbox" className="ml-2 text-sm font-medium text-gray-900 dark:text-gray-300">
                I agree with
                {' '}
                <span className="underline">Terms</span>
                {' '}
                and
                {' '}
                <span className="underline">Privacy</span>
              </label>
              <div className="text-green-700 text-xs ml-2">
                {errors.accept_terms?.message}
              </div>
            </div>
          </div>
          <div className="mb-4">
            <Button className="w-full" text="Sign Up" type="submit" />
          </div>
          <p className="text-center text-gray-500 text-xs">
            Already a user ?
            {' '}
            <Link to="/login" className="underline">Sign In</Link>
          </p>
          <p className="text-center text-gray-500 text-xs">OR</p>
          <p className="text-center text-gray-500 text-xs">
            Go back
            {' '}
            {' '}
            <Link to="/" className="underline">Home</Link>
          </p>
        </form>
      </div>
    </div>
  );
};

export default SignUp;
