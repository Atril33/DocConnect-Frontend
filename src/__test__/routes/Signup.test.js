import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';
import { Provider } from 'react-redux';
import store from '../../redux/store';
import SignUp from '../../routes/SignUp';

test('Login component renders correctly', () => {
  const queryClient = new QueryClient();
  const component = renderer.create(
    <Router>
      <QueryClientProvider client={queryClient}>
        <Provider store={store}>
          <SignUp />
        </Provider>
      </QueryClientProvider>
    </Router>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
