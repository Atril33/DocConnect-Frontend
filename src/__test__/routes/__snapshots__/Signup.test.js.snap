// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUP component renders correctly 1`] = `
<div
  className="bg-home-background-mobile bg-cover bg-center h-screen relative flex justify-center items-center md:bg-home-background"
>
  <div
    className="w-full max-w-xs"
  >
    <h1
      className="text-white text-center mb-6 text-5xl font-bold"
    >
      Sign Up
    </h1>
    <form
      className="bg-white shadow-md rounded px-8 pt-6 pb-8 mb-4"
      onSubmit={[Function]}
    >
      <div
        className="mb-4"
      >
        <div>
          <div
            className="relative h-10 w-full min-w-[200px]"
          >
            <input
              className="peer h-full w-full rounded-[7px] border border-gray-400 border-t-transparent bg-transparent px-3 py-2.5 font-sans text-sm font-normal text-blue-gray-700 outline outline-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-gray-400 focus:border-2 focus:border-green-400 focus:border-t-transparent focus:outline-0 disabled:border-0 disabled:bg-blue-gray-50"
              name="name"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=" "
              type="text"
            />
            <label
              className="before:content[' '] after:content[' '] pointer-events-none absolute left-0 -top-1.5 flex h-full w-full select-none text-[11px] font-normal leading-tight text-blue-gray-400 transition-all before:pointer-events-none before:mt-[6.5px] before:mr-1 before:box-border before:block before:h-1.5 before:w-2.5 before:rounded-tl-md before:border-t before:border-l before:border-gray-400 before:transition-all after:pointer-events-none after:mt-[6.5px] after:ml-1 after:box-border after:block after:h-1.5 after:w-2.5 after:flex-grow after:rounded-tr-md after:border-t after:border-r after:border-gray-400 after:transition-all peer-placeholder-shown:text-sm peer-placeholder-shown:leading-[3.75] peer-placeholder-shown:text-blue-gray-500 peer-placeholder-shown:before:border-transparent peer-placeholder-shown:after:border-transparent peer-focus:text-[11px] peer-focus:leading-tight peer-focus:text-green-400 peer-focus:before:border-t-2 peer-focus:before:border-l-2 peer-focus:before:border-green-400 peer-focus:after:border-t-2 peer-focus:after:border-r-2 peer-focus:after:border-green-400 peer-disabled:text-transparent peer-disabled:before:border-transparent peer-disabled:after:border-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500"
            >
              Username
            </label>
          </div>
        </div>
        <div
          className="text-green-700 text-xs"
        />
      </div>
      <div
        className="mb-4"
      >
        <div>
          <div
            className="relative h-10 w-full min-w-[200px]"
          >
            <input
              className="peer h-full w-full rounded-[7px] border border-gray-400 border-t-transparent bg-transparent px-3 py-2.5 font-sans text-sm font-normal text-blue-gray-700 outline outline-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-gray-400 focus:border-2 focus:border-green-400 focus:border-t-transparent focus:outline-0 disabled:border-0 disabled:bg-blue-gray-50"
              name="email"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=" "
              type="email"
            />
            <label
              className="before:content[' '] after:content[' '] pointer-events-none absolute left-0 -top-1.5 flex h-full w-full select-none text-[11px] font-normal leading-tight text-blue-gray-400 transition-all before:pointer-events-none before:mt-[6.5px] before:mr-1 before:box-border before:block before:h-1.5 before:w-2.5 before:rounded-tl-md before:border-t before:border-l before:border-gray-400 before:transition-all after:pointer-events-none after:mt-[6.5px] after:ml-1 after:box-border after:block after:h-1.5 after:w-2.5 after:flex-grow after:rounded-tr-md after:border-t after:border-r after:border-gray-400 after:transition-all peer-placeholder-shown:text-sm peer-placeholder-shown:leading-[3.75] peer-placeholder-shown:text-blue-gray-500 peer-placeholder-shown:before:border-transparent peer-placeholder-shown:after:border-transparent peer-focus:text-[11px] peer-focus:leading-tight peer-focus:text-green-400 peer-focus:before:border-t-2 peer-focus:before:border-l-2 peer-focus:before:border-green-400 peer-focus:after:border-t-2 peer-focus:after:border-r-2 peer-focus:after:border-green-400 peer-disabled:text-transparent peer-disabled:before:border-transparent peer-disabled:after:border-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500"
            >
              Email
            </label>
          </div>
        </div>
        <div
          className="text-green-700 text-xs"
        />
      </div>
      <div
        className="mb-4"
      >
        <div>
          <div
            className="relative h-10 w-full min-w-[200px]"
          >
            <input
              className="peer h-full w-full rounded-[7px] border border-gray-400 border-t-transparent bg-transparent px-3 py-2.5 font-sans text-sm font-normal text-blue-gray-700 outline outline-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-gray-400 focus:border-2 focus:border-green-400 focus:border-t-transparent focus:outline-0 disabled:border-0 disabled:bg-blue-gray-50"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=" "
              type="password"
            />
            <label
              className="before:content[' '] after:content[' '] pointer-events-none absolute left-0 -top-1.5 flex h-full w-full select-none text-[11px] font-normal leading-tight text-blue-gray-400 transition-all before:pointer-events-none before:mt-[6.5px] before:mr-1 before:box-border before:block before:h-1.5 before:w-2.5 before:rounded-tl-md before:border-t before:border-l before:border-gray-400 before:transition-all after:pointer-events-none after:mt-[6.5px] after:ml-1 after:box-border after:block after:h-1.5 after:w-2.5 after:flex-grow after:rounded-tr-md after:border-t after:border-r after:border-gray-400 after:transition-all peer-placeholder-shown:text-sm peer-placeholder-shown:leading-[3.75] peer-placeholder-shown:text-blue-gray-500 peer-placeholder-shown:before:border-transparent peer-placeholder-shown:after:border-transparent peer-focus:text-[11px] peer-focus:leading-tight peer-focus:text-green-400 peer-focus:before:border-t-2 peer-focus:before:border-l-2 peer-focus:before:border-green-400 peer-focus:after:border-t-2 peer-focus:after:border-r-2 peer-focus:after:border-green-400 peer-disabled:text-transparent peer-disabled:before:border-transparent peer-disabled:after:border-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500"
            >
              Password
            </label>
          </div>
        </div>
        <div
          className="text-green-700 text-xs"
        />
      </div>
      <div
        className="mb-6"
      >
        <div>
          <div
            className="relative h-10 w-full min-w-[200px]"
          >
            <input
              className="peer h-full w-full rounded-[7px] border border-gray-400 border-t-transparent bg-transparent px-3 py-2.5 font-sans text-sm font-normal text-blue-gray-700 outline outline-0 transition-all placeholder-shown:border placeholder-shown:border-blue-gray-200 placeholder-shown:border-t-gray-400 focus:border-2 focus:border-green-400 focus:border-t-transparent focus:outline-0 disabled:border-0 disabled:bg-blue-gray-50"
              name="password_confirmation"
              onBlur={[Function]}
              onChange={[Function]}
              placeholder=" "
              type="password"
            />
            <label
              className="before:content[' '] after:content[' '] pointer-events-none absolute left-0 -top-1.5 flex h-full w-full select-none text-[11px] font-normal leading-tight text-blue-gray-400 transition-all before:pointer-events-none before:mt-[6.5px] before:mr-1 before:box-border before:block before:h-1.5 before:w-2.5 before:rounded-tl-md before:border-t before:border-l before:border-gray-400 before:transition-all after:pointer-events-none after:mt-[6.5px] after:ml-1 after:box-border after:block after:h-1.5 after:w-2.5 after:flex-grow after:rounded-tr-md after:border-t after:border-r after:border-gray-400 after:transition-all peer-placeholder-shown:text-sm peer-placeholder-shown:leading-[3.75] peer-placeholder-shown:text-blue-gray-500 peer-placeholder-shown:before:border-transparent peer-placeholder-shown:after:border-transparent peer-focus:text-[11px] peer-focus:leading-tight peer-focus:text-green-400 peer-focus:before:border-t-2 peer-focus:before:border-l-2 peer-focus:before:border-green-400 peer-focus:after:border-t-2 peer-focus:after:border-r-2 peer-focus:after:border-green-400 peer-disabled:text-transparent peer-disabled:before:border-transparent peer-disabled:after:border-transparent peer-disabled:peer-placeholder-shown:text-blue-gray-500"
            >
              Confirm password
            </label>
          </div>
        </div>
        <div
          className="text-green-700 text-xs"
        />
      </div>
      <div
        className="flex items-center mb-4"
      >
        <input
          checked={false}
          className="w-4 h-4 text-blue-600 bg-gray-100 border-gray-300 rounded focus:ring-blue-500 dark:focus:ring-blue-600 dark:ring-offset-gray-800 focus:ring-2 dark:bg-gray-700 dark:border-gray-600"
          id="checked-checkbox"
          name="accept_terms"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
        />
        <div>
          <label
            className="ml-2 text-sm font-medium text-gray-900 dark:text-gray-300"
            htmlFor="checked-checkbox"
          >
            I agree with
             
            <span
              className="underline"
            >
              Terms
            </span>
             
            and
             
            <span
              className="underline"
            >
              Privacy
            </span>
          </label>
          <div
            className="text-green-700 text-xs ml-2"
          />
        </div>
      </div>
      <div
        className="mb-4"
      >
        <button
          className="bg-green-400 hover:bg-green-700 text-white font-bold py-2 px-4 rounded w-full"
          type="submit"
        >
          Sign Up
        </button>
      </div>
      <p
        className="text-center text-gray-500 text-xs"
      >
        Already a user ?
         
        <a
          className="underline"
          href="/login"
          onClick={[Function]}
        >
          Sign In
        </a>
      </p>
      <p
        className="text-center text-gray-500 text-xs"
      >
        OR
      </p>
      <p
        className="text-center text-gray-500 text-xs"
      >
        Go back
         
         
        <a
          className="underline"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </p>
    </form>
  </div>
</div>
`;
